fvwm/builtins.c

```diff
@@ -2385,6 +2385,55 @@ void CMD_CursorMove(F_CMD_ARGS)
 	return;
 }
 
+void CMD_CursorSet(F_CMD_ARGS)
+{
+	int val1, val2, val1_unit, val2_unit;
+	int vx, vy;
+	if (GetTwoArguments(action, &val1, &val2, &val1_unit, &val2_unit) != 2)
+	{
+		fvwm_debug(__func__, "CursorSet needs 2 arguments");
+		return;
+	}
+	int x = val1 * val1_unit / 100;
+	int y = val2 * val2_unit / 100;
+	unsigned int w = 0, h = 0;
+	struct monitor *m = NULL;
+	TAILQ_FOREACH(m, &monitor_q, entry) {
+		w = m->virtual_scr.MyDisplayWidth;
+		h = m->virtual_scr.MyDisplayHeight;
```

Comment:
This needs correcting, as it's been a long-standing semantic bug.  `m->virtual_scr.MyDisplayWidth` and `m->virtual_scr.MyDisplayHeight` are not per-monitor values, but an aggregate of all displays connected.  Hence, you either want to set `w`/`h` to `monitor_get_all_widths()`/`monitor_get_all_heights()` respectively outside of the loop, or use `m->si->w`/`m->si->h` for per-monitor values.

I'll clean this up separately.